### ISSUE TITLE
Add a `CMakeLists.txt` to the library tarball for downstream CMake projects

### ISSUE THREAD

**Issue by LTLA** [usnistgov/h5wasm#10]
The static libraries requested previously (#8) work like a charm, and have been quite easy to integrate into my existing CMake project. Perhaps you may be interested in the `CMakeLists.txt` I was using for this. This idea is to put this file inside the `libhdf5_wasm.tgz` tarball to allow CMake to do a configure-time fetch of **h5wasm**, eliminating the need for developers to manually manage this particular dependency. (Non-CMake users are unaffected and can just ignore the extra file.)

My `CMakeLists.txt` file is available at https://github.com/LTLA/h5wasm-cmake-demo; downstream packages can then just do:

```
FetchContent_Declare(
  h5wasm
  GIT_REPOSITORY https://github.com/LTLA/h5wasm-cmake-demo
  GIT_TAG master
)
FetchContent_MakeAvailable(h5wasm)
```

which exposes `h5wasm` and `h5wasm_cpp` as library targets. A similar approach can be used for a tarball, something like:

```
FetchContent_Declare(
  h5wasm
  URL https://github.com/usnistgov/h5wasm/releases/download/v0.2.0/libhdf5_wasm.tgz
  URL_HASH MD5=<whatever-the-hash-is>
)
```

You can see a test example of the former approach at https://github.com/LTLA/bind-h5wasm-test, with a more real-life use case at jkanche/scran.js#34. Some more elbow grease could be applied to do proper exports so that the exposed targets look like `h5wasm::h5wasm`, but I don't know enough about CMake to say whether that's sensible in this case. 

Of course, downstream packages could just define all the targets themselves if there wasn't a `CMakeLists.txt` in the tarball, but this is a bit of a chore. It seems worthwhile to have a single definition inside the tarball for immediate consumption via something like:

```
target_link_libraries(mylib h5wasm_cpp)
```

**Comment by bmaranville** [comment]
It seems like a separate repo just for building libhdf5 might be useful - I had wondered about that in the past.  Then h5wasm could depend on it the way you describe here.

**Comment by bmaranville** [comment]
Thanks for the contribution of the CMakeLists.txt, also!  The C++ code you have written for scran.js is very clean and readable - it makes me wonder if I should rewrite the h5wasm C library in C++.

**Comment by LTLA** [comment]
> It seems like a separate repo just for building libhdf5 might be useful - I had wondered about that in the past. Then h5wasm could depend on it the way you describe here.

Yes, this is also the approach used by Bioconductor's [**Rhdf5lib**](https://bioconductor.org/packages/release/bioc/html/Rhdf5lib.html) package, which provides the HDF5 libraries for the actual R interface in [**rhdf5**](https://bioconductor.org/packages/release/bioc/html/rhdf5.html) package. This seems to work pretty well at satistfying developers at both the R and C/C++ level.

> Thanks for the contribution of the CMakeLists.txt, also! The C++ code you have written for scran.js is very clean and readable - it makes me wonder if I should rewrite the h5wasm C library in C++.

My pleasure! HDF5's C++ API is pretty intuitive... but I will mention that the documentation seems relatively poor compared to that of the C API. I had to spend some time staring at C++ method signatures and comparing them to the corresponding page of the C docs to figure out what I'm meant to do. Also, the inherent C'ness of the underlying libraries leaks through the C++ interface every now and then, e.g., global identifiers in property list construction, incompatibility with C++11 parallelization.

**Comment by bmaranville** [comment]
Thanks for all your help with this - I just released [v0.1.0](https://github.com/usnistgov/libhdf5-wasm/releases/tag/v0.1.0) (there will be very few releases) at https://github.com/usnistgov/libhdf5-wasm - it provides libhdf5.a etc. for three different versions of HDF5, including an embedded version of the CMakeLists.txt you provided.
